integration/devmode/devmode_test.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package devmode
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"syscall"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/hyperledger/fabric/integration/nwo"
+	"github.com/hyperledger/fabric/integration/nwo/commands"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"github.com/tedsuo/ifrit"
+	"github.com/tedsuo/ifrit/ginkgomon"
+)
+
+var _ = Describe("Devmode", func() {
+	var (
+		testDir                  string
+		client                   *docker.Client
+		network                  *nwo.Network
+		process                  ifrit.Process
+		chaincode                nwo.Chaincode
+		chaincodeRunner          *ginkgomon.Runner
+		chaincodeProcess         ifrit.Process
+		channelName, chaincodeID string
+	)
+
+	BeforeEach(func() {
+		var err error
+		channelName = "testchannel"
+		testDir, err = ioutil.TempDir("", "devmode")
+		Expect(err).NotTo(HaveOccurred())
+
+		client, err = docker.NewClientFromEnv()
+		Expect(err).NotTo(HaveOccurred())
+
+		chaincode = nwo.Chaincode{
+			Name:    "mycc",
+			Version: "0.0",
+			Path:    "github.com/hyperledger/fabric/integration/chaincode/simple/cmd",
+			Ctor:    `{"Args":["init","a","100","b","200"]}`,
+			Policy:  `AND ('Org1MSP.member','Org2MSP.member')`,
+		}
+
+		chaincodeID = fmt.Sprintf("%s:%s", chaincode.Name, chaincode.Version)
+		network = nwo.New(nwo.BasicSolo(), testDir, client, BasePort(), components)
+		network.TLSEnabled = false
+		// turn on dev mode for all peers
+		for _, p := range network.Peers {
+			p.DevMode = true
+		}
+
+		network.GenerateConfigTree()
+		network.Bootstrap()
+
+		networkRunner := network.NetworkGroupRunner()
+		process = ifrit.Invoke(networkRunner)
+		Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+
+	AfterEach(func() {
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+
+		if chaincodeProcess != nil {
+			chaincodeProcess.Signal(syscall.SIGTERM)
+		}
+
+		if network != nil {
+			network.Cleanup()
+		}
+
+		os.RemoveAll(testDir)
+	})
+
+	It("executes chaincode in dev mode", func() {
+		org1peer0 := network.Peer("Org1", "peer0")
+		orderer := network.Orderer("orderer")
+
+		By("setting up the channel")
+		network.CreateAndJoinChannel(orderer, channelName)
+
+		By("building the chaincode")
+		chaincodePath, err := gexec.Build(chaincode.Path)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("running the chaincode")
+		peerChaincodeAddress := network.PeerAddress(org1peer0, nwo.ChaincodePort)
+		envs := []string{fmt.Sprintf("CORE_PEER_ADDRESS=%s", peerChaincodeAddress), fmt.Sprintf("CORE_CHAINCODE_ID_NAME=%s", chaincodeID)}
+		cmd := exec.Command(chaincodePath)
+		cmd.Env = append(cmd.Env, envs...)
+		chaincodeRunner = ginkgomon.New(ginkgomon.Config{
```

Comment:
This might seem like an odd question, but - why are you running the chaincode in a separate process? Can't you run the chaincode directly from within this test? You'll have better control of it, and by taking control from the test you will ensure that it is your chaincode that is connecting to the peer.